src/shared/machine-id-setup.c

```diff
@@ -72,7 +78,7 @@ static int acquire_machine_id(const char *root, sd_id128_t *ret) {
 
         if (empty_or_root(root) && running_in_chroot() <= 0) {
                 /* Let's use a system credential for the machine ID if we can */
-                if (acquire_machine_id_from_credential(ret) >= 0)
+                if (acquire_machine_id_from_credential(ret, &machine_id_from_firmware) >= 0)
```

Comment:
the above would then mean you have to store the results in a local variable first here, and then copy them over.

Reply:
Not 100% sure I understood you correctly, but I tried to implement this like you told me.